### ISSUE TITLE
Why SDK set IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED for MQTT messages not ACKed?

### ISSUE THREAD

**Issue by xiongyu0523** [Azure/azure-iot-sdk-c#2037]
I'm trying to understand why the SDK report IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED error in connection status callback for telemetry that not being ACKed. 

The name of this reason tells it has reached a max. number of **connection** retires, before reach this limit, the sdk should perform several times of retry first. Is my understanding correct?

```
static void ProcessPendingTelemetryMessages(PMQTTTRANSPORT_HANDLE_DATA transport_data)
{
    PDLIST_ENTRY current_entry = transport_data->telemetry_waitingForAck.Flink;
    tickcounter_ms_t current_ms;
    (void)tickcounter_get_current_ms(transport_data->msgTickCounter, &current_ms);
    while (current_entry != &transport_data->telemetry_waitingForAck)
    {
        MQTT_MESSAGE_DETAILS_LIST* msg_detail_entry = containingRecord(current_entry, MQTT_MESSAGE_DETAILS_LIST, entry);
        DLIST_ENTRY nextListEntry;
        nextListEntry.Flink = current_entry->Flink;

        if (((current_ms - msg_detail_entry->msgPublishTime) / 1000) > RESEND_TIMEOUT_VALUE_MIN)
        {
            if (msg_detail_entry->retryCount >= MAX_SEND_RECOUNT_LIMIT)
            {
                notifyApplicationOfSendMessageComplete(msg_detail_entry->iotHubMessageEntry, transport_data, IOTHUB_CLIENT_CONFIRMATION_MESSAGE_TIMEOUT);
                (void)DList_RemoveEntryList(current_entry);
                free(msg_detail_entry);

                DisconnectFromClient(transport_data);
                if (!transport_data->isRetryExpiredCallbackCalled) // Only call once
                {
                    transport_data->transport_callbacks.connection_status_cb(IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED, IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED, transport_data->transport_ctx);
                    transport_data->isRetryExpiredCallbackCalled = true;
                }
```

**Comment by ericwol-msft** [comment]
#2070

**Comment by ericwol-msft** [comment]
If a message does not received a PUBACK, we will call SendMessageComplete callback with a status of `IOTHUB_CLIENT_CONFIRMATION_MESSAGE_TIMEOUT`

**Comment by ericwol-msft** [comment]
#2081 2078

**Comment by ericwol-msft** [comment]
resolved with PR #2078
